test/cases/container-runtime-defaults.sh

```diff
@@ -96,3 +96,25 @@ if [[ ${exit_code} -ne 0 ]]; then
     echo "❌ Test Failed: expected a zero exit code but got '${exit_code}'"
     exit 1
 fi
+
+echo "--> Should fail when given an invalid containerd config"
+config_file="$TEMP_DIR/containerd.toml"
+cat > "$config_file" <<EOF
+[cgroup]
+path = "foo"
+[cgroup]
+path = "bar"
+EOF
+
+export KUBELET_VERSION=v1.24.15-eks-ba74326
+/etc/eks/bootstrap.sh \
+    --b64-cluster-ca dGVzdA== \
+    --apiserver-endpoint http://my-api-endpoint \
+    --container-runtime containerd \
+    --containerd-config-file "$config_file" \
+    test || exit_code=$?
+
+if [[ ${exit_code} -eq 0 ]]; then
+    echo "❌ Test Failed: expected a non-zero exit code but got '${exit_code}'"
+    exit 1
+fi
```

Comment:
We should move this test case to a different file, since it's not related to container runtime defaults. Additionally, it'd be great to add a test for the validity of the containterd config file we have in `files/`.

e.g. something like `containerd-config.sh`:
```
echo "--> Default containerd config file should be valid"
mv /etc/eks/containerd/containerd-config.toml /etc/containerd/config.toml
STDERR_FILE=$(mktemp)
containerd config dump >/dev/null 2>$STDERR_FILE || exit_code=$?

if [[ ${exit_code} -ne 0 ]]; then
    echo "❌ Test Failed: default containerd config file is invalid! $(cat $STDERR_FILE)"
    exit 1
fi

echo "--> Should fail when given an invalid containerd config"
config_file="$TEMP_DIR/containerd.toml"
cat > "$config_file" <<EOF
[cgroup]
path = "foo"
[cgroup]
path = "bar"
EOF

export KUBELET_VERSION=v1.24.15-eks-ba74326
/etc/eks/bootstrap.sh \
    --b64-cluster-ca dGVzdA== \
    --apiserver-endpoint http://my-api-endpoint \
    --container-runtime containerd \
    --containerd-config-file "$config_file" \
    test || exit_code=$?

if [[ ${exit_code} -eq 0 ]]; then
    echo "❌ Test Failed: expected a non-zero exit code but got '${exit_code}'"
    exit 1
fi
```

Reply:
Added. FWIW the happy path is also tested with the "Should allow containerd as container runtime when below k8s version 1.24" case.

```
--> Should allow containerd as container runtime when below k8s version 1.24
Using kubelet version 1.20.15
Using containerd as the container runtime
'/etc/eks/containerd/containerd-config.toml' -> '/etc/containerd/config.toml'
'/etc/eks/containerd/sandbox-image.service' -> '/etc/systemd/system/sandbox-image.service'
'/etc/eks/containerd/kubelet-containerd.service' -> '/etc/systemd/system/kubelet.service'
containerd: failed to load TOML: /etc/containerd/config.toml: (3, 1): parsing error: keys cannot contain new lines
Exited with error on line 537
❌ Test Failed: expected a zero exit code but got '1'
❌ ❌ container-runtime-defaults.sh Tests Failed! ❌ ❌
```